tools/docker/dev_container.Dockerfile

```diff
@@ -1,67 +1,89 @@
 #syntax=docker/dockerfile:1.1.5-experimental
 ARG IMAGE_TYPE
+ARG TF_VERSION
 ARG PY_VERSION
+ARG TF_NEED_CUDA
+ARG TF_NAME
 ARG HOROVOD_VERSION
+ARG BUILD_IMAGE
+ARG PROTOBUF_VERSION
 
 # Currenly all of our dev images are GPU capable but at a cost of being quite large.
 # See https://github.com/tensorflow/build/pull/47
-FROM tensorflow/build:latest-python$PY_VERSION as dev_container
-ARG TF_PACKAGE
-ARG TF_VERSION
+FROM ${BUILD_IMAGE} as dev_container
+
+RUN echo "#! /usr/bin/python2.7" >> /usr/bin/lsb_release2
+RUN cat /usr/bin/lsb_release >> /usr/bin/lsb_release2
+RUN mv /usr/bin/lsb_release2 /usr/bin/lsb_release
+
 ARG PY_VERSION
+RUN ln -sf /usr/local/bin/python$PY_VERSION /usr/bin/python
 
-RUN pip uninstall $TF_PACKAGE -y
-RUN pip install --default-timeout=1000 $TF_PACKAGE==$TF_VERSION
+ENV PATH=/dt8/usr/bin:${PATH}
+ENV LD_LIBRARY_PATH=/usr/local/lib:${LD_LIBRARY_PATH}
+ENV LD_LIBRARY_PATH=/dt8/user/lib64:${LD_LIBRARY_PATH}
+ENV LD_LIBRARY_PATH=/dt8/user/lib:${LD_LIBRARY_PATH}
+ENV MANPATH=/dt8/user/share/man:${LD_LIBRARY_PATH}
+ENV INFOPATH=/dt8/user/share/info
 
-RUN rm -rf /usr/lib/python3
-RUN rm -rf /usr/lib/python
-RUN ln -sf /usr/lib/python$PY_VERSION /usr/lib/python
-RUN ln -sf /usr/lib/python$PY_VERSION /usr/lib/python3
+ARG TF_VERSION
+ARG TF_NAME
+ARG HOROVOD_VERSION
+ARG PROTOBUF_VERSION
 
 RUN python -m pip install --upgrade pip
+RUN python -m pip install --default-timeout=1000 $TF_NAME==$TF_VERSION
 
-COPY tools/install_deps /install_deps
-COPY requirements.txt /tmp/requirements.txt
-RUN pip install -r /install_deps/yapf.txt \
-    -r /install_deps/typedapi.txt \
-    -r /tmp/requirements.txt
+COPY tools/docker/install/install_horovod.sh /install/
+RUN /install/install_horovod.sh $HOROVOD_VERSION
 
-RUN pip install setuptools
+COPY tools/install_deps/ ./
+COPY tools/docker/install/install_pytest.sh /install/
+RUN bash /install/install_pytest.sh
 
-RUN bash /install_deps/buildifier.sh
-RUN bash /install_deps/clang-format.sh
+COPY requirements.txt .
+RUN python -m pip install -r requirements.txt
 
-COPY tools/docker/install/install_bazel.sh /install/
-RUN /install/install_bazel.sh "5.1.1"
+RUN python -m pip install tensorflow-io
 
-ENV ADDONS_DEV_CONTAINER="1"
+RUN python -m pip install --upgrade protobuf==$PROTOBUF_VERSION
 
-RUN apt-get update && apt-get install -y \
-      openssh-client \
-      cmake
+COPY tools/install_deps/yapf.txt ./
+RUN pip install -r ./yapf.txt
 
-RUN apt-get update && apt-get remove -y python3-apt \
-      && apt-get install -y python3-apt
+RUN pip install setuptools
 
-COPY tools/docker/install/install_openmpi.sh /install/
-RUN /install/install_openmpi.sh "4.1.1"
+COPY tools/install_deps/buildifier.sh ./buildifier.sh
+RUN bash buildifier.sh
 
-COPY tools/docker/install/install_nccl.sh /install/
-RUN /install/install_nccl.sh "2.8.4-1+cuda11.2"
+COPY tools/install_deps/clang-format.sh ./clang-format.sh
```

Comment:
Hi @jq , just updated to add this clang-format installation, may you have time to review and approve it? Thanks!